day04/spec/unit/main_spec.rb

```diff
@@ -0,0 +1,94 @@
+require_relative '../../lib/main.rb'
+
+require 'pry'
+RSpec.describe Main do
+
+
+  let(:main) {
+    Main.new
+  }
+
+
+
+  context ".choose_product" do
+
+    it "returns a product" do
+      allow(main).to receive(:gets).and_return("Pen")
+      #binding.pry
+      expect(main.choose_product).to be_a_kind_of(Hash)
+    end
+
+    it "choose existing product" do
+      allow(main).to receive(:gets).and_return("Pen")
+      #binding.pry
+      expect(main.choose_product).to be_a_kind_of(Hash)
+      expect(main.choose_product[:name]).to eq("Pen")
+    end
+
+  end
+
+  context ".add_product_to_basket" do
+
+    it "add product from warehouse" do
+      double1 = double
+      allow(main).to receive(:gets).and_return(double1)
+      allow(double1).to receive(:chomp).and_return("Pen", 2)
+      expect{main.add_product_to_basket}.to change{main.get_basket.size}.from(0).to(1)
+    end
+
+    it "add product from basket" do
+
+    end
```

Comment:
If you skip `do... end` block and just leave `it "adds product from basket"` when you run rspec it will be shown in the list of specs as pending. This may be a reminder to implement it and not leave it blank.